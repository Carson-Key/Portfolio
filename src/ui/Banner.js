import { MdMenu } from 'react-icons/md'
import Logo512 from '../assets/Logo-512.png'

const Header = (props) => {
	const { openDrawerMenu } = props

	return (
		<header className="flex relative w-screen h-14 z-40 text-white bg-secondary">
			<button onClick={openDrawerMenu}><MdMenu className="absolute inset-y-0 left-0 h-14 text-2xl ml-2"/></button>
			<div className="flex justify-center my-auto mx-auto">
				<h1 className="my-auto text-lg font-helvetica">Carson</h1>
				<img src={Logo512} className="h-10 ml-1" alt="" />
				<h1 className="my-auto text-lg font-helvetica ml-1">Key</h1>
			</div>
		</header>
	)
}

export default Header
