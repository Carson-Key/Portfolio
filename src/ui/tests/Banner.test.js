// Packages
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
// UI
import Banner from '../Banner'

describe('Banner Component', () => {
    it('Should Render Banner Component', () => {
        const banner = renderer.create(
            <Router>
                <Banner />
            </Router>
        ).toJSON()
        expect(banner).toMatchSnapshot()
    })
})