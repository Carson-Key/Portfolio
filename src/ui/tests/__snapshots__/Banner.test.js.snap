// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component Should Render Banner Component 1`] = `
Array [
  <button
    aria-hidden="true"
    aria-label="Mobile Navigation Button"
    className="lg:hidden"
  >
    <svg
      className="absolute inset-y-0 left-0 h-14 text-2xl ml-2"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>,
  <a
    aria-current="page"
    className="flex justify-center my-auto mx-auto lg:mx-3 active"
    href="/"
    onClick={[Function]}
  >
    <h1
      aria-label="Title first name"
      className="my-auto text-lg font-helvetica lg:text-2xl xl:text-3xl 2xl:text-4xl"
    >
      Carson
    </h1>
    <img
      alt="Brass colored Node.js hexagon logo with brass colored React logo inside, which has the nucleus replaced with a brass key "
      className="h-10 ml-1 lg:h-14 xl:h-16 2xl:h-20"
      src="Logo-512.png"
    />
    <h1
      aria-label="Title last name"
      className="my-auto text-lg font-helvetica ml-1 lg:text-2xl xl:text-3xl 2xl:text-4xl"
    >
      Key
    </h1>
  </a>,
]
`;
